File: pys/paxnetnews.py
import requests
from bs4 import BeautifulSoup

url = 'https://paxnetnews.com/categories'

req = requests.get(url)
bs = BeautifulSoup(req.text, 'lxml')

wrapper = bs.find('div', class_='list')
items = wrapper.find_all('div', class_='list-article')

for item in items[:1]:
    div = item.find('div', class_='content')
    a_tag = div.find('a', class_='dyn std')
    title = a_tag.get_text().strip()
    href = 'https://paxnetnews.com' + a_tag['href']
    date = item.find('div', class_='pubdate').get_text().strip()

    req2 = requests.get(href)
    bs2 = BeautifulSoup(req2.content, 'lxml')
    wrapper2 = bs2.find('article', class_='article')

    contents = ''.join([it.get_text().strip() for it in wrapper2.find_all('p')]).replace('\n', '')

    print(title, href, date, contents)

File: crawlers/paxnetnews.go
package crawlers

import (
	"fmt"
	"net/http"

	"flashnews/models"
	"flashnews/utils"

	"github.com/PuerkitoBio/goquery"
)

const paxnetNewsCommonURL = "https://paxnetnews.com/categories"
const paxnetNewsItemURL = "https://paxnetnews.com"

type PaxnetNews struct{}

func (c PaxnetNews) GetName() string {
	return "paxnetnews"
}

func (c PaxnetNews) GetList(number int) ([]models.NewsItem, error) {
	// Number
	var _number int
	if number > 15 || number < 1 {
		_number = 15
	} else {
		_number = number
	}
	result := make([]models.NewsItem, _number)

	// Request
	req, err := http.Get(paxnetNewsCommonURL)
	if err != nil {
		return result, err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return result, fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return result, err
	}

	// Parsing
	wrapper := html.Find("div.list")
	items := wrapper.Find("div.list-article")
	items.Each(func(i int, sel *goquery.Selection) {
		if i >= _number {
			return
		}

		div := sel.Find("div.content")
		aTag := div.Find("a.dyn.std")
		href, ok := aTag.Attr("href")
		if !ok {
			result[i] = models.NewsItem{}
			return
		}
		url := paxnetNewsItemURL + href

		date := utils.TrimAll(sel.Find("div.pubdate").Text())
		title := aTag.Text()

		result[i] = models.NewsItem{
			Title:    title,
			URL:      url,
			Contents: "",
			Datetime: date,
		}
	})

	return result, nil
}

func (c PaxnetNews) GetContents(item *models.NewsItem) error {
	// Request
	req, err := http.Get(item.URL)
	if err != nil {
		return err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return err
	}

	// Parsing
	wrapper := html.Find("article.article")
	contents := ""
	wrapper.Find("p").Each(func(idx int, sel *goquery.Selection) {
		contents += (utils.TrimAll(sel.Text()))
	})

	item.Contents = contents
	return nil
}
